contracts/utils/RepAllocation.sol

```diff
@@ -42,6 +44,48 @@ contract RepAllocation is Ownable {
         }
     }
 
+    /**
+     * @dev add addBeneficiariesRoot function
+     * @param _root Merkle Tree root
+     * @param _beneficiaries user addresses
+     * @param _amounts allocations
+     */
+    function addBeneficiariesRoot(
+        bytes32 _root,
+        address[] memory _beneficiaries,
+        uint256[] memory _amounts
+    ) public onlyOwner {
+        require(!reputationAllocationsRoots[_root]);
+        reputationAllocationsRoots[_root] = true;
+        emit BeneficiaryAddressAddedByRoot(_root, _beneficiaries, _amounts);
+    }
+
+    /**
+     * @dev add addBeneficiariesRoot function
+     * https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/cryptography/MerkleProof.sol
+     * @param _beneficiary user address
+     * @param _amount allocation
+     * @param _proof Merkle proof related to previously submitted Merkle root
+     */
+    function revealBeneficiary(address _beneficiary, uint256 _amount, bytes32[] memory _proof) public {
```

Comment:
how about using openzeppline MarkleProof lib verify function ? 
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/cryptography/MerkleProof.sol#L13

Reply:
It can be used, I used my own because think it will be useful to recover `index` for invalidation. Now I see invalidation is done inside `addBeneficiary`.

Reply:
But you should also add root argument, because OZ implementation only verifies root matches, do not return recovered root.

Reply:
Fixed